### ISSUE TITLE
package organization is schizophrenic with demeteorizer 3 / meteor 1.0.5

### ISSUE THREAD

**Issue by jedwards1211** [onmodulus/demeteorizer#158]
When I use demeteorizer 3 to package a Meteor 1.0.5 project, some of the deps wind up in the `bundle/packages/server/npm` instead of `package.json` or `npm-shrinkwrap.json`.  This happens to `bcrypt`, for example, whereas with demeteorizer 2 / meteor 1.0.5, `bcrypt` winds up in `package.json` like everything else.

Is this intended behavior, or does demeteorizer behave differently with Meteor 1.2?  I thought all deps were supposed to be managed by npm in the bundle, rather than some being pre-packaged in an `npm` folder.

**Comment by sjmcdowall** [comment]
demeteorizer 3 now uses the native "meteor build" script to gather the piece parts -- and I think meteor build changed from 1.0 to 1.2, which may explain the situation above.  Can you upgrade the project to meteor 1.2.X?

**Comment by jedwards1211** [comment]
Last I checked Meteor 1.2 was still having issues taking forever to build large JS bundles.  If that's changed we can upgrade...in any case perhaps there should be a note in the readme that <1.2 users should use Demeteorizer 2?

**Comment by jedwards1211** [comment]
@sjmcdowall Even when I'm using Demeteorizer 2.3.1 with Meteor 1.0.5...the output bundle still contains a `programs/server/npm` folder, and `programs/server/program.json` still contains references to modules inside the `npm` folder, for example `npm-bcrypt`.

Isn't demeteorizer supposed to get rid of the `npm` folder entirely so that all `main.js` loads all dependencies from `node_modules`?

I don't get what the point is otherwise.  When I tried running the demeteorized project on a different device I got an invalid ELF header error because it was loading `bcrypt` from the `npm` folder instead of from `node_modules/bcrypt` which `npm i` built specifically for that device.

**Comment by jedwards1211** [comment]
@sjmcdowall sorry, nevermind, the subfolders of `npm` are empty...not sure what's wrong yet

**Comment by fiveisprime** [comment]
This is the intended behavior. As @sjmcdowall mentioned, demeteorizer now uses
the built-in meteor build tool rather than attempting to generate a
package.json.

Demeteorizer version 2 tried to build up the pacakge.json, which worked 95% of
the time; Demeteorizer 3 uses `meteor build` to generate the bundle then ensures
that the necessary pieces are present in pakcage.json (node version and start
script). This works about 99% of the time. The generated bundle is organized in
a way that the modules in the `npm` will be correctly resolved when used within
the application code.

In other words, it's all good. Don't worry too much about it - the bundle may
differ between versions of Meteor because of changes to the build tool, but has
been very reliable regardless of how ugly it may end up. :+1:
